src/librustdoc/html/static/main.js

```diff
@@ -1173,100 +1171,25 @@ function hideThemeButtonState() {
             });
         }
 
-        var func = function(e) {
-            var next = e.nextElementSibling;
-            if (next && hasClass(next, "item-info")) {
-              next = next.nextElementSibling;
-            }
-            if (!next) {
-                return;
-            }
-            if (hasClass(next, "docblock")) {
-                var newToggle = toggle.cloneNode(true);
-                insertAfter(newToggle, e.childNodes[e.childNodes.length - 1]);
-                if (hideMethodDocs === true && hasClass(e, "method") === true) {
-                    collapseDocs(newToggle, "hide");
+        if (hideMethodDocs === true) {
+            onEachLazy(document.getElementsByClassName("method"), function(e) {
+                var toggle = e.parentNode;
+                if (toggle) {
+                    toggle = toggle.parentNode;
                 }
-            }
-        };
-
-        var funcImpl = function(e) {
-            var next = e.nextElementSibling;
-            if (next && hasClass(next, "item-info")) {
-                next = next.nextElementSibling;
-            }
-            if (next && hasClass(next, "docblock")) {
-                next = next.nextElementSibling;
-            }
-            if (!next) {
-                return;
-            }
-        };
+                if (toggle && toggle.tagName === "DETAILS") {
+                    toggle.open = false;
+                }
+            });
+        }
```

Comment:
This would fit in more nicely as part of the `onEachLazy(document.getElementsByTagName("details"), function (e) {` loop below. That sections finds all the appropriate details tags and opens them if needed, which means we don't have to make the JS dependent on a very specific DOM structure. Since methods are default-open, we would add a clause to say:

```
var hideMethod = hideMethods && hasClass(e, "methods-toggle");
if (hideMethod) {
  e.open = false;
}
```

Note that we also need the class `methods-toggle` in the emitted HTML.

Reply:
So I'm shared here: I'm not a big fan of the idea to grow the HTML, but that would indeed improve the JS. So after giving it some thought, here is what I suggest: we will move the "method"/"function"/etc classes from the `h3`/`h4` into the grand-parent details. However, it seems to be out of context here, so if you're fine with it: I'll open an issue so that it can be done later and not be forgotten.

Reply:
I agree HTML size is important, but I don't think we should economize by omitted classes when they are needed. That makes the CSS and JS more complex and harder to reason about. I'd rather look to bigger potential wins first - like deduplicating the tooltips subdocuments.

There's also an argument from consistency: We already put appropriate classes on the `<details>` tags we've added so far. It would be odd to economize on this one particular item type and not the others.

At any rate, I don't want to hold up the whole PR for the next week; I'm find landing this piece as-is and we can discuss more when I'm back.